examples/puppetmaster/vagrant/windows2016/README.md

```diff
@@ -0,0 +1,190 @@
+# Vagrant File for Creating a Windows 2016 Virtual Machine with Puppet Agent
+
+#### Table of Contents
+
+1. [Description](#description)
+2. [Setup](#setup)
+    * [Setup requirements](#setup-requirements)
+    * [Clone this Repository](#clone-this-repository)
+    * [Setting up Vagrant](#setting-up-vagrant)
+    * [Setting up VirtualBox](#setting-up-virtualbox)
+3. [Create a Puppet Master](#create-a-puppet-master)
+4. [Create a Windows2016 VM](#create-a-windows2016-vm)
+5. [Install Puppet Agent on the Windows VM](install-puppet-agent-on-the-windows-vm)
+6. [Create an /etc/hosts Entry for Conjur in Puppet Master Container](#create-an-etc-hosts-entry-for conjur-in-puppet-master-container)
+7. [Run Puppet Agent on the Windows VM](install-puppet-agent-on-the-windows-vm)
+
+## Description
+
+This directory contains a Vagrant file for creating a Windows 2016 virtual
+machine (VM) running on VirtualBox, and starting up a Puppet agent service
+on that VM.
+
+The Windows 2016 VM with Puppet agent can be run in conjunction with the
+[puppetmaster example](../../docker-compose.yml) in this repository, which
+provides a Puppet server running in a Docker container.
+
+### Overview
+
+- Mac/Linux host with Vagrant and VirtualBox installed
+- Puppet server running locally via docker-compose (using examples/puppetmaster/smoketest_e2e.sh)
+- "Vanilla" Windows2016 VM created via Vagrant
+- Installing/Running Puppet Agent
+    - Agent installation and run can be done remotely from the Mac via `vagrant powershell -c <command>`.
+    - Puppet Agent can be installed/replaced dynamically, so devs should be able to keep a long-standing
+      "vanilla" Windows VM (powered down when not in use), and install/replace Puppet agent as needed.
+
+#### Puppet and Conjur Servers
+
+- The setup is based on examples/puppetmaster/smoketest_e2e.sh, modified to:
+   - Don't do cleanup after testing Linux agents
+   - Expose Conjur on host port 30301
+   - Expose Puppet on host port 30302
+   - In Puppet server container, add /etc/hosts entry `<docker-gateway-ip> conjur`
+
+#### Puppet Agent
+
+- Puppet agent is dynamically installed/replaced on Windows VM
+- Agent version is selectable
+- Pre-provisioned machine identity using regedit and windows credentials manager
+- Follow instructions below to install/run Puppet agent
+
+## Setup
+
+### Setup Requirements
+
+This module requires that you have:
+- Oracle VirtualBox, Version 6.0 or later (installation instructions
+  [here](https://docs.oracle.com/en/virtualization/virtualbox/6.1/user/installation.html))
+- Vagrant, Version 2.2.9 or later (installation instructions
+  [here](https://www.vagrantup.com/docs/installation))
+- 60 GB of free disk space on your development server
+
+### Clone this Repository
+
+Make a local copy of this repository as follows:
+
+```sh-session
+    git clone https://github.com/cyberark/conjur-puppet
+    cd examples/puppetmaster/vagrant/windows2016_puppet
+```
+
+### Setting up Vagrant
```

Comment:
This doesn't say how to install vagrant, just how to install the plugin

Reply:
I added the link to the Vagrant installation instructions to this section.